core/src/main/java/com/linecorp/armeria/client/retry/BackoffSpec.java

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.retry;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import com.google.common.base.Ascii;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.MoreObjects.ToStringHelper;
+import com.google.common.base.Splitter;
+
+/**
+ * A specification of a {@link Backoff} configuration represented by a string. The string syntax is
+ * a series of comma-separated options(key-values pair) and each values is colon-separated.
+ * The options are divided into two groups which are base options and decorators. Base options include
+ * {@link ExponentialBackoff}, {@link FixedBackoff} and {@link RandomBackoff}. Only one of them can be
+ * specified in the {@code specification}, otherwise {@link IllegalArgumentException} will occur.
+ * <ul>
+ *     <li>{@code exponential=[initialDelayMillis:maxDelayMillis:multiplier]}
+ *     is for {@link Backoff#exponential(long, long, double)}
+ *     (maxDelayMillis and multiplier can be omitted)</li>
+ *     <li>{@code fixed=[delayMillis]} is for {@link Backoff#fixed(long)}</li>
+ *     <li>{@code random=[minDelayMillis:maxDelayMillis]} is for {@link Backoff#random(long, long)}</li>
+ *     <li>{@code jitter=[minJitterRate:maxJitterRate]} is for {@link Backoff#withJitter(double, double)}
+ *     (if only one jitter value is specified, it will be used for {@link Backoff#withJitter(double)}</li>
+ *     <li>{@code maxAttempts=[maxAttempts]} is for {@link Backoff#withMaxAttempts(int)}</li>
+ * </ul>
+ * So for example, you can specify it such as {@code random=100:10000,jitter=-0.4:0.3,maxAttempts=100}.
+ * The order of options does not matter, and the {@code specification} needs at least an option. If you don't
+ * specify them, the values are set by default values. The default values are
+ * {@code exponential=200:10000:2.0,jitter=0.2,maxAttempts=10}. After the values are set by
+ * {@link #parse(String)}, you can create the {@link Backoff} by calling {@link #build()}.
+ */
+final class BackoffSpec {
```

Comment:
I think he named it after `CaffeineSpec`. I'm not strong on this. Builder sounds fine as well.